src/Pages/HomePage/HomePage.js

```diff
@@ -41,15 +99,43 @@ export function HomePage() {
          <div className='video-info'>270k Views |  8 months ago  <span className='options'><i className="fa-solid fa-ellipsis-vertical options-icon" onClick={() => toggleBtn(video._id)}></i></span></div>
          </div>
          <div className='video-model-btns' id={video.toggle ? 'visible' : 'hidden' }>
-           <div className='video-model-btn'><i className="fa-solid fa-clock-rotate-left"></i> Add To Watch Later</div>
-           <div className='video-model-btn'><i className="fa-solid fa-list-check"></i> Add To Playlist</div>
+         {
+           video.watchLater
+           ?  <div className='video-model-btn' onClick={() => removeVideoFromWatchLater(video)}><i className="fa-solid fa-clock-rotate-left"></i> Remove From Watch Later</div>
+           :  <div className='video-model-btn' onClick={() => addVideoToWatchLater(video)}><i className="fa-solid fa-clock"></i> Add To Watch Later</div>
+         }
+           <div className='video-model-btn' onClick={() => showPlaylistModal(video._id)}><i className="fa-solid fa-list-check"></i> Add To Playlist</div>
 
          </div>
+         <div className={showModal ? 'playlistModal' : 'hidden'}>
+           <h2 className='modal-heading'>Create Playlist</h2>
+           <div>
+           <input 
+           type="text" 
+           placeholder='Playlist Name...' 
+           className='modal-input' 
+           value={playlistName} 
+           onChange={(e) => setPlaylistName(e.target.value)} />
+
+           <button className='playlist-btn' onClick={createPlaylist}>Create Playlist</button>
+           </div>
+           <button className="close-btn dismiss-btn" onClick={closeModal}>X</button>
+           {playlists.map(playlist => (
+             <div className='playlist-container' key={playlist._id}>
+               <input type="checkbox"
+               onChange={(e) => updatePlayList(e, playlist._id, video)} />
+               <div>{playlist.title}</div>
+             </div>
+           ))}
+       </div>
        </div>
      ))}
      </div>
+     {/* modal section */} 
+     
     </div>
     </div>
+    
     </div>
   )
 }
```

Comment:
Suggestion on Homepage you can use ellipsis and wrap video title to one line so that all video cards remains of same size